js/site-nav.js

```diff
@@ -47,21 +49,33 @@ SiteNav.prototype.hide = function() {
   this.isOpen = false;
 };
 
-SiteNav.prototype.handleChange = function(e) {
-  var $subToggle = $(e.currentTarget);
-  var checked = $subToggle.is(':checked');
-  var $list = $(e.currentTarget).siblings('ul');
+SiteNav.prototype.toggleSublist = function(e) {
+  var method = this.$openSublist ? this.hideSublist : this.showSublist;
+  var $target = $(e.target);
+  var $sublistParent = $target.hasClass('js-sublist-parent') ? $target : $target.closest('.js-sublist-parent');
+  method.call(this, $sublistParent);
+};
 
-  $list.attr('aria-hidden', !checked);
-  $list.find('li a').first().focus();
+SiteNav.prototype.showSublist = function($sublistParent) {
+  $sublistParent.addClass('is-open');
+  $sublistParent.find('ul').attr('aria-hidden', false);
+  this.$openSublist = $sublistParent;
 };
 
-SiteNav.prototype.handleMouseEnter = function(e) {
-  $(e.currentTarget).find('.js-toggle').prop('checked', true).change();
+SiteNav.prototype.hideSublist = function($sublistParent) {
+  $sublistParent.removeClass('is-open');
+  $sublistParent.find('ul').attr('aria-hidden', true);
+  this.$openSublist = null;
 };
 
-SiteNav.prototype.handleMouseExit = function(e) {
-  $(e.currentTarget).find('.js-toggle').prop('checked', false).change();
+SiteNav.prototype.handleFocus = function(e) {
+  var $target = $(e.target);
+  var $sublistParent = $target.closest('.js-sublist-parent') || false;
```

Comment:
The `|| false` part probably isn't necessary here. We only get to that branch if the value of `closest(...)` is false-y, and in that case, we already have a false-y value to be checked on line 76.